sup3r/preprocessing/batch_handling.py

```diff
@@ -155,11 +253,203 @@ def get_coarse_batch(cls, high_res,
         low_res = smooth_data(low_res, training_features, smoothing_ignore,
                               smoothing)
         high_res = cls.reduce_features(high_res, output_features_ind)
-        batch = cls(low_res, high_res)
+        output = cls.make_output(low_res, high_res,
+                                 s_enhance, t_enhance,
+                                 model_mom1, output_features_ind)
+        mask = cls.make_mask(high_res,
+                             s_padding, t_padding)
+        batch = cls(low_res, high_res, output, mask)
 
         return batch
 
 
+class BatchMom2(Batch):
+    """Batch of low_res, high_res and output data
+    when learning second moment"""
+
+    @staticmethod
+    def make_output(low_res, high_res,
+                    s_enhance=None, t_enhance=None,
+                    model_mom1=None, output_features_ind=None):
+        """Make custom batch output
+
+        Parameters
+        ----------
+        low_res : np.ndarray
+            4D | 5D array
+            (batch_size, spatial_1, spatial_2, features)
+            (batch_size, spatial_1, spatial_2, temporal, features)
+        high_res : np.ndarray
+            4D | 5D array
+            (batch_size, spatial_1, spatial_2, features)
+            (batch_size, spatial_1, spatial_2, temporal, features)
+        s_enhance : int | None
+            Spatial enhancement factor
+        t_enhance : int | None
+            Temporal enhancement factor
+        model_mom1 : Sup3rCondMom | None
+            Model used to modify the make the batch output
+        output_features_ind : list | np.ndarray | None
+            List/array of feature channel indices that are used for generative
+            output, without any feature indices used only for training.
+        """
+        # Remove first moment from HR and square it
+        out = model_mom1._tf_generate(low_res).numpy()
+        return (high_res - out)**2
+
+
+class BatchMom2Sep(Batch):
```

Comment:
This could also just call Batch.make_output()**2

Reply:
This one might make the code less readable, no?

Reply:
Well I think Batch should be subclassed as BatchMom1Sep (or something like that) with make_output returning just high_res and then BatchMom2Sep could just return BatchMom1Sep.make_output**2, which I think reads better than having the same repeated doc string.

Reply:
Even better would be a single n-th conditional moment Batch class that is completely general, which just takes n as an input. But maybe this isn't feasible.